### DISCUSSION TITLE
revamping variants or creating a new one

### DISCUSSION THREAD

**Discussion by MuhammadJamaluddin** [system-ui/theme-ui#1901] [Q&A]
How could I remap built-in variants? for example, the `IconButton` variants are defined in the `theme.buttons` object, what if I wanted to define them in a separate `theme.iconButton`?

& what about non-built-in components as well? Let's say I have a component `Component` & I want to define its variants within `theme.componentVariants`... how can I do that?

**Comment by lachlanjc** [comment]
Remapping internal variants is not supported; for that custom a use case you’ll need to build your own components.

For your own components, you can make them accept a variant prop, then render

```js
variant={`componentVariants.${props.variant}`}
```
